provider/provider_nodejs_test.go

```diff
@@ -63,3 +63,18 @@ func TestRegress3094(t *testing.T) {
 	upResult := test.Up()
 	t.Logf("#%v", upResult.Summary)
 }
+
+func TestRegress3835(t *testing.T) {
+	skipIfShort(t)
+	dir := filepath.Join("test-programs", "regress-3835")
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+	providerName := "aws"
+	options := []opttest.Option{
+		opttest.LocalProviderPath(providerName, filepath.Join(cwd, "..", "bin")),
+		opttest.YarnLink("@pulumi/aws"),
+	}
+	test := pulumitest.NewPulumiTest(t, dir, options...)
+	upResult := test.Up()
```

Comment:
I feel like we should test `preview` instead here because based on the integration test failure, this is actually invalid code that will fail on deployment. Seems like the real issue is https://github.com/pulumi/pulumi-terraform-bridge/pull/1881 causing the preview value to be incorrect.

Reply:
Good point. It reproduces on preview.